File: src/main/java/com/github/kindrat/cassandra/client/ui/window/menu/file/AuthCredentialsBox.java
package com.github.kindrat.cassandra.client.ui.window.menu.file;

import com.github.kindrat.cassandra.client.i18n.MessageByLocaleService;
import com.github.kindrat.cassandra.client.properties.UIProperties;
import javafx.geometry.Pos;
import javafx.scene.control.TextField;
import javafx.scene.layout.VBox;

public class AuthCredentialsBox extends VBox {
    private final TextField username;
    private final TextField password;
    private final MessageByLocaleService localeService;

    AuthCredentialsBox(MessageByLocaleService localeService, UIProperties uiProperties) {
        this.localeService = localeService;
        username = getUsernameField();
        password = getPasswordField();
        getChildren().add(username);
        getChildren().add(password);
        setHeight(uiProperties.getCredentialsBoxHeight());
        setSpacing(uiProperties.getCredentialsBoxSpacing());
        setVisible(false);
    }

    public String getUsername() {
        return username.getText();
    }

    public String getPassword() {
        return password.getText();
    }

    @Override
    protected void setWidth(double value) {
        super.setWidth(value);
        username.setMinWidth(value);
        username.setMaxWidth(value);
        username.setPrefWidth(value);
        password.setMinWidth(value);
        password.setMaxWidth(value);
        password.setPrefWidth(value);
    }

    private TextField getUsernameField() {
        TextField username = new TextField();
        username.setPromptText(localeService.getMessage("ui.menu.file.connect.auth.username"));
        username.setAlignment(Pos.CENTER);
        return username;
    }

    private TextField getPasswordField() {
        TextField username = new TextField();
        username.setPromptText(localeService.getMessage("ui.menu.file.connect.auth.password"));
        username.setAlignment(Pos.CENTER);
        return username;
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/window/menu/file/AuthCredentialsBox.kt
package com.github.kindrat.cassandra.client.ui.window.menu.file

import com.github.kindrat.cassandra.client.i18n.MessageByLocaleService
import com.github.kindrat.cassandra.client.properties.UIProperties
import javafx.geometry.Pos
import javafx.scene.control.TextField
import javafx.scene.layout.VBox

class AuthCredentialsBox internal constructor(
        private val localeService: MessageByLocaleService,
        uiProperties: UIProperties
) : VBox() {

    private val username: TextField
    private val password: TextField

    fun getUsername(): String {
        return username.text
    }

    fun getPassword(): String {
        return password.text
    }

    fun setUsername(userId: String?) {
        username.text = userId
    }

    fun setPassword(passwd: String?) {
        password.text = passwd
    }

    override fun setWidth(value: Double) {
        super.setWidth(value)
        username.minWidth = value
        username.maxWidth = value
        username.prefWidth = value
        password.minWidth = value
        password.maxWidth = value
        password.prefWidth = value
    }

    private val usernameField: TextField
        get() {
            val username = TextField()
            username.promptText = localeService.getMessage("ui.menu.file.connect.auth.username")
            username.alignment = Pos.CENTER
            return username
        }

    private val passwordField: TextField
        get() {
            val username = TextField()
            username.promptText = localeService.getMessage("ui.menu.file.connect.auth.password")
            username.alignment = Pos.CENTER
            return username
        }

    init {
        username = usernameField
        password = passwordField
        children.add(username)
        children.add(password)
        height = uiProperties.credentialsBoxHeight
        spacing = uiProperties.credentialsBoxSpacing
        isVisible = false
    }
}